### ISSUE TITLE
How I can see mAP value of Yolov5 on Raspberry Pi?

### ISSUE THREAD

**Issue by jaskiratsingh2000** [karanjakhar/yolov5-export-to-raspberry-pi#1]
@karanjakhar How I can see mAP value of Yolov5 on Raspberry Pi?

**Comment by karanjakhar** [comment]
I will add that option. Please wait.

**Comment by jaskiratsingh2000** [comment]
Hey @karanjakhar Did you added this functionality? Can I test that? 
Well, I am just a researcher working in this domain

**Comment by jaskiratsingh2000** [comment]
Hey @karanjakhar I am still waiting on to hear from you. Please let me know

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Why are you not replying? Can you please let me know. I feel that having open ended conversation is very important for amy competition. 
@karanjakhar If you don't have any answer to this, then also let me know so that I shouldn't bother you up.

Thanks

**Comment by karanjakhar** [comment]
Hi @jaskiratsingh2000  I am occupied with other work. I am afraid I can't work on it till this weekend. Sorry to keep you waiting.

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Yes please that would be great because this is very important. Unless there is correct accuracy there is no use of Export

**Comment by jaskiratsingh2000** [comment]
Hey @karanjakhar As you were mentioning to work on this on weekend so did you get a chance to work on it? Has it is implemented?

**Comment by karanjakhar** [comment]
@jaskiratsingh2000 I got the vaccine at weekend, and it was rough :) 
It's on my list bro.

**Comment by jaskiratsingh2000** [comment]
Sure no worries @karanjakhar Take care. I am pinging you again and again because I am a research intern working on the project and I took your export version as the reference hence it is very important for me to understand the mAP(Accuracy Performance) of Yolo on Raspberry Pi because with the original YOLO I am facing this issue. Do you have any idea about this issue? I am sure this will be helpful for you as well to look since you would also be testing and applying on your version and until you get a clear idea of why that is happening on the original version then I think it would be hard for you as well to move forward.
Here is the link to my problem which I am trying ti reference you again and again - https://docs.google.com/document/d/1ej8hIktTFmsqTH0GCYVw-qD8sF9C9fa_Pe61EenNmkU/edit?usp=drivesdk

So @karanjakhar please please let me know since I would really appreciate your response on this

**Comment by karanjakhar** [comment]
@jaskiratsingh2000 I will look into it also.

**Comment by jaskiratsingh2000** [comment]
Thank you so much! I appreciate that :)
Looking forward to hearing from you.

Thanks and Regards

**Comment by jaskiratsingh2000** [comment]
Hi, @karanjakhar Did you get a chance to look into the above? I am still waiting to hear back from you at least on a doc.

**Comment by jaskiratsingh2000** [comment]
@karanjakhar did you get a chance to look into it? Please let me know as I am just waiting on to hear back from you and won't be able to move forward ahead.

**Comment by jaskiratsingh2000** [comment]
@karanjakhar I am still waiting on your reply. Please let me know that if you get a chance to look into the above and work on mAP thing?

**Comment by jaskiratsingh2000** [comment]
Hey @karanjakhar It's been quite a while now and I haven't heard from you. Please please let me know. Would really appreciate your response.

Also, I am unable to download Tensorflow 2.4.1 as per the requirement how can I do that?

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Please reply. I have been pinging you for since long. Would really appreciate your response. Please let me know

**Comment by jaskiratsingh2000** [comment]
Hey Its been so many days that I haven't heard from you. You actually
mentioned that you would be working on this and my ahead things are just
waiting on you since this is the reference I used.
Please please let me know how long can I expect to figure out the mAP value?

**Comment by karanjakhar** [comment]
Apologies for the late reply. I have been occupied with work. This month I am committed to other work, I will try if I get some time to implement the mAP calculation but I can't commit anything. 

I looked into the doc. Please do not train any model just use pre-trained weights on both rasp pi 4 and the server machine and update me what is the result.

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Thanks for the reply. This was the first thing I did and I also did this again and below is the output I got:

```
pubgcur@raspberryhthythypi:~/Desktop/yolov5 $ python3 val.py --data coco128.yaml --weights yolov5s.pt                        
Downloading https://ultralytics.com/assets/Arial.ttf to /tmp/Ultralytics/Arial.ttf...                            
100%|█████████████████████████████████████████████████████████████████████████| 755k/755k [00:00<00:00, 2.13MB/s]│
val: data=./data/coco128.yaml, weights=['yolov5s.pt'], batch_size=32, imgsz=640, conf_thres=0.001, iou_thres=0.6,
 task=val, device=, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=·
False, save_json=False, project=runs/val, name=exp, exist_ok=False, half=False                                   
YOLOv5 🚀 v5.0-419-gc5360f6 torch 1.9.0a0+gitd69c22d CPU                                                         
                                                                                                                
Downloading https://github.com/ultralytics/yolov5/releases/download/v5.0/yolov5s.pt to yolov5s.pt...             
100%|███████████████████████████████████████████████████████████████████████| 14.1M/14.1M [00:03<00:00, 4.43MB/s]
                                                                                                                 
Fusing layers...                                                                                                 
Model Summary: 224 layers, 7266973 parameters, 0 gradients                                                      
                                                                                                                 
WARNING: Dataset not found, nonexistent paths: ['/home/pi/Desktop/datasets/coco128/images/train2017']            
Downloading https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip ...                         
100%|███████████████████████████████████████████████████████████████████████| 6.66M/6.66M [00:01<00:00, 4.44MB/s]
Dataset autodownload success                                                                                     
                                                                                                               
val: Scanning '../datasets/coco128/labels/train2017' images and labels...128 found, 0 missing, 2 empty, 0 corrupt
val: New cache created: ../datasets/coco128/labels/train2017.cache                                               
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|█| 4/4 [04:14<00:00,
                 all        128        929     0.0151    0.00428   0.000619   6.19e-05                           ·
Speed: 13.8ms pre-process, 1929.2ms inference, 13.1ms NMS per image at shape (32, 3, 640, 640)                   ·
Results saved to runs/val/exp
```

@karanjakhar This is without training. Please let me know what should I do? That is why I was after your repo so that it should have the mAP mechanism as well

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Are you there?? Please let me know

**Comment by karanjakhar** [comment]
Please also run the same command on other machine and share the output.

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Here is the output I got from the running on server:

```
Downloading https://ultralytics.com/assets/Arial.ttf to /tmp/Ultralytics/Arial.ttf...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 755k/755k [00:00<00:00, 26.0MB/s]
val: data=./data/coco128.yaml, weights=['yolov5s.pt'], batch_size=32, imgsz=640, conf_thres=0.001, iou_thres=0.6, task=val, device=, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=exp, exist_ok=False, half=False
YOLOv5 🚀 v5.0-419-gc5360f6 torch 1.9.0+cu102 CUDA:0 (NVIDIA GeForce RTX 2080 Ti, 11019.4375MB)

Downloading https://github.com/ultralytics/yolov5/releases/download/v5.0/yolov5s.pt to yolov5s.pt...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 14.1M/14.1M [00:00<00:00, 17.2MB/s]

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients

WARNING: Dataset not found, nonexistent paths: ['/home/arani/datasets/coco128/images/train2017']
Downloading https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip ...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 6.66M/6.66M [00:00<00:00, 12.4MB/s]
Dataset autodownload success

val: Scanning '../datasets/coco128/labels/train2017' images and labels...128 found, 0 missing, 2 empty, 0 corrupted: 100%|█| 128/128 [00:00<00:0
val: New cache created: ../datasets/coco128/labels/train2017.cache
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|█████████████████████| 4/4 [00:13<00:00,  3.40s/it]
                 all        128        929      0.698      0.564      0.643      0.425
Speed: 0.4ms pre-process, 14.8ms inference, 56.0ms NMS per image at shape (32, 3, 640, 640)
Results saved to runs/val/exp
```

**Comment by jaskiratsingh2000** [comment]
@karanjakhar and this is what I got on my local Ubuntu Machine:

```
Downloading https://ultralytics.com/assets/Arial.ttf to /tmp/Ultralytics/Arial.ttf...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 755k/755k [00:00<00:00, 2.24MB/s]
val: data=./data/coco128.yaml, weights=['yolov5s.pt'], batch_size=32, imgsz=640, conf_thres=0.001, iou_thres=0.6, task=val, device=, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=exp, exist_ok=False, half=False
YOLOv5 🚀 v5.0-419-gc5360f6 torch 1.9.0+cu102 CPU

Downloading https://github.com/ultralytics/yolov5/releases/download/v5.0/yolov5s.pt to yolov5s.pt...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 14.1M/14.1M [00:04<00:00, 3.50MB/s]

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients

WARNING: Dataset not found, nonexistent paths: ['/home/jaskirat/Research-IIITDelhi/datasets/coco128/images/train2017']
Downloading https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip ...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 6.66M/6.66M [00:01<00:00, 4.24MB/s]
Dataset autodownload success

val: Scanning '../datasets/coco128/labels/train2017' images and labels...128 found, 0 missing, 2 empty, 0 corrupted: 100%|█| 128/128 [00:00<00:0
val: New cache created: ../datasets/coco128/labels/train2017.cache
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|█████████████████████| 4/4 [01:10<00:00, 17.74s/it]
                 all        128        929      0.666      0.585      0.645      0.427
Speed: 4.4ms pre-process, 518.4ms inference, 13.9ms NMS per image at shape (32, 3, 640, 640)
Results saved to runs/val/exp
```

**Comment by jaskiratsingh2000** [comment]
@karanjakhar I am just wondering how did you test this while you were preparing this export repo?

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Did you get a chance to look into the above?

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Did you get a chance to look into the above?

**Comment by jaskiratsingh2000** [comment]
Hey when will you reply?

**Comment by jaskiratsingh2000** [comment]
@karanjakhar Its been so long and you haven't answered me yet. Can you please let me know that if you read my doc and got a chance to work on the mAP value?
